### ISSUE TITLE
Default Select mapping doesn't work on Mac

### ISSUE THREAD

**Issue by bodgit** [snesrev/zelda3#217]
### Describe your bug here. And how to reproduce it.

Compiled on Mac no problem. Game runs fine however Select mapped to Right Shift doesn't work, tried Left Shift too. Mapping to some random key like Q works so it seems to be something specific to reading the Shift keys.

### What is your build target?

Mac

**Comment by bodgit** [comment]
With a bit of poking about in the source I managed to get it working with the following change:
```diff
diff --git a/config.c b/config.c
index ff3e6a1..d06d64e 100644
--- a/config.c
+++ b/config.c
@@ -111,7 +111,8 @@ int FindCmdForSdlKey(SDL_Keycode code, SDL_Keymod mod) {
     return 0;
   int key = mod & KMOD_ALT ? kKeyMod_Alt : 0;
   key |= mod & KMOD_CTRL ? kKeyMod_Ctrl : 0;
-  key |= mod & KMOD_SHIFT ? kKeyMod_Shift : 0;
+  if (code != SDLK_LSHIFT && code != SDLK_RSHIFT)
+    key |= mod & KMOD_SHIFT ? kKeyMod_Shift : 0;
   key |= REMAP_SDL_KEYCODE(code);
   return KeyMapHash_Find(key);
 }
```
So only set the shift modifier if the key isn't Left or Right Shift. This fixes the Select mapping and shifted combinations like Shift+W still seem to work.

I'm not sure what the correct SDL behaviour should be? If the key pressed is Shift, should the modifier also be set? FYI I have SDL 2.26.1 installed which is only one patch release behind the latest.

**Comment by edbefee3-3888-462a-9411-741b7e9eb54e** [comment]
Doesn't work on Linux either

**Comment by bodgit** [comment]
> Doesn't work on Linux either

That's reassuring, I wasn't sure if I had bumped into something Mac-specific, although SDL should be abstracting all of this away.

Does the above patch fix it for you?

I'm wondering if the Ctrl and Alt code is also broken if you want to use them as keys rather than just modifiers.

**Comment by bodgit** [comment]
> I'm wondering if the Ctrl and Alt code is also broken if you want to use them as keys rather than just modifiers.

Yup, seems to be. I applied the same tweak to Ctrl and Alt too:
```diff
diff --git a/config.c b/config.c
index ff3e6a1..dbf689a 100644
--- a/config.c
+++ b/config.c
@@ -109,9 +109,13 @@ static int KeyMapHash_Find(uint16 key) {
 int FindCmdForSdlKey(SDL_Keycode code, SDL_Keymod mod) {
   if (code & ~(SDLK_SCANCODE_MASK | 0x1ff))
     return 0;
-  int key = mod & KMOD_ALT ? kKeyMod_Alt : 0;
-  key |= mod & KMOD_CTRL ? kKeyMod_Ctrl : 0;
-  key |= mod & KMOD_SHIFT ? kKeyMod_Shift : 0;
+  int key = 0;
+  if (code != SDLK_LALT && code != SDLK_RALT)
+    key |=  mod & KMOD_ALT ? kKeyMod_Alt : 0;
+  if (code != SDLK_LCTRL && code != SDLK_RCTRL)
+    key |= mod & KMOD_CTRL ? kKeyMod_Ctrl : 0;
+  if (code != SDLK_LSHIFT && code != SDLK_RSHIFT)
+    key |= mod & KMOD_SHIFT ? kKeyMod_Shift : 0;
   key |= REMAP_SDL_KEYCODE(code);
   return KeyMapHash_Find(key);
 }
```

**Comment by snesrev** [comment]
Fixed in aefcb08
